src/components/Priority/Priority.js

```diff
@@ -1,116 +1,188 @@
-  },
-})
+    borderRadius: 10
+  }
+});
 
 const marksWithLabels = [
   {
     value: 0,
-    label: 'low',
+    label: "low"
   },
   {
     value: 25,
-    label: '',
+    label: ""
   },
   {
     value: 50,
-    label: 'medium',
+    label: "medium"
   },
   {
     value: 75,
-    label: '',
+    label: ""
   },
   {
     value: 100,
-    label: 'high',
-  },
-]
+    label: "high"
+  }
+];
 
 const marksWithoutLabels = [
   {
     value: 0,
-    label: '',
+    label: ""
   },
   {
     value: 25,
-    label: '',
+    label: ""
   },
   {
     value: 50,
-    label: '',
+    label: ""
   },
   {
     value: 75,
-    label: '',
+    label: ""
   },
   {
     value: 100,
-    label: '',
-  },
-]
+    label: ""
+  }
+];
 
 function valuetext(value) {
   return `${value}%`;
 }
-
-function Priority({ goalAddress, onClose }) {
-
-  const classes = useStyles()
+function Priority({
+  goalAddress,
+  onClose,
+  createGoalVote,
+  whoami,
+  fetchGoalVotes,
+  updateGoalVote
+}) {
+  const classes = useStyles();
 
   const priorityItemVars = [
-    { "priorityIcon": "urgency.svg", "priorityItemTitle": "Urgency" },
-    { "priorityIcon": "importance.svg", "priorityItemTitle": "Importance" },
-    { "priorityIcon": "impact.svg", "priorityItemTitle": "Impact" },
-    { "priorityIcon": "effort.png", "priorityItemTitle": "Effort" },
-  ]
-  const priorityItems = priorityItemVars.map((priorityItem, index) =>
-    <div className="priority_item">
-      <Icon className='priority_item_icon' name={priorityItem.priorityIcon} size='small' />
-      <div className="priority_item_title">{priorityItem.priorityItemTitle}</div>
+    { priorityIcon: "urgency.svg", priorityItemTitle: "Urgency" },
+    { priorityIcon: "importance.svg", priorityItemTitle: "Importance" },
+    { priorityIcon: "impact.svg", priorityItemTitle: "Impact" },
+    { priorityIcon: "effort.png", priorityItemTitle: "Effort" }
+  ];
+  const [value, setValue] = useState({
+    Urgency: 50,
+    Importance: 50,
+    Impact: 50,
+    Effort: 50
+  });
+  const hadleOnChange = e => {
+    if (
+      value[e.target.parentElement.parentElement.children[1].textContent] !==
+      e.target.children[0].children[0].children[0].textContent
+    ) {
+      console.log(
+        "object",
+        e.target.children[0].children[0].children[0].textContent
+      );
+      setValue({
+        ...value,
+        [e.target.parentElement.parentElement.children[1].textContent]:
+          e.target.children[0].children[0].children[0].textContent
+      });
+    }
+  };
+  useEffect(() => {
+    console.log("state", value);
+    let goal_vote = {
+      goal_address: goalAddress,
+      urgency: parseInt(value["Urgency"]) / 100 || 0,
+      importance: parseInt(value["Importance"]) / 100 || 0,
+      impact: value["Impact"] / 100,
+      effort: parseInt(value["Effort"]) / 100 || 0,
+      agent_address: whoami.entry.address,
+      unix_timestamp: Date.now()
+    };
+    fetchGoalVotes().then(value => {
```

Comment:
Instead of calling `fetchGoalVotes` here... call it along with this code instead:
https://github.com/h-be/acorn-ui/blob/d0445431b8f8ca37373355dcd040a9463f6011c8/src/index.js#L50-L55